.github/workflows/e2e-tests.yaml

```diff
@@ -90,3 +90,59 @@ jobs:
           retention-days: 10
           path: |
             ${{ env.KUBERAY_TEST_OUTPUT_DIR }}/**/*.log
+
+  ray-kubectl-plugin:
+        runs-on: ubuntu-20.04
+        steps:
+          - name: Checkout code
+            uses: actions/checkout@v3
+            with:
+              submodules: recursive
+
+          - name: Set up Go
+            uses: actions/setup-go@v3
+            with:
+              go-version: v1.22
+
+          - name: Set up gotestfmt
+            uses: gotesttools/gotestfmt-action@v2
+            with:
+              token: ${{ secrets.GITHUB_TOKEN }}
+
+          - name: Setup and start KinD cluster
+            uses: ./.github/workflows/actions/kind
+
+          - name: Build CLI and Add to PATH
+            run: |
+              cd ./kubectl-plugin
+
+              go mod download
+              go build -o kubectl-ray -a ./cmd/kubectl-ray.go
+
+              cp ./kubectl-ray /usr/local/bin
+
+          - name: Deploy Kuberay operator
+            id: deploy
+            run: |
+              echo Deploying Kuberay operator
+              cd ray-operator
+
+              IMG="${REGISTRY_ADDRESS}"/kuberay
+              make docker-build -e IMG="${IMG}" -e ENGINE=podman
+              make docker-push -e IMG="${IMG}" -e ENGINE=podman
+
+              make deploy -e IMG="${IMG}"
+              kubectl wait --timeout=90s --for=condition=Available=true deployment -n ray-system kuberay-operator
+
+          - name: Run e2e tests
+            run: |
+              export KUBERAY_TEST_TIMEOUT_SHORT=1m
+              export KUBERAY_TEST_TIMEOUT_MEDIUM=5m
+              export KUBERAY_TEST_TIMEOUT_LONG=10m
+
+              export KUBERAY_TEST_OUTPUT_DIR=${{ env.TEMP_DIR }}
+              echo "KUBERAY_TEST_OUTPUT_DIR=${KUBERAY_TEST_OUTPUT_DIR}" >> $GITHUB_ENV
+
+              set -euo pipefail
+              cd kubectl-plugin
+              go test -timeout 30m -v ./test/e2e -json 2>&1 | tee ${KUBERAY_TEST_OUTPUT_DIR}/gotest.log | gotestfmt
```

Comment:
These lines duplicate a lot with

https://github.com/ray-project/kuberay/blob/dea87fff2c3d83800f5f35df73bfb3e8b7adcfb1/.github/workflows/e2e-tests.yaml#L30-L92

Consider refactor them using [Reusable workflows](https://docs.github.com/en/actions/using-workflows/reusing-workflows).

Reply:
Hi, @MortalHappiness can you take a look at what I currently have? Still currently figuring things out and would appreciate for some reason the reusable workflow ".github/workflows/e2e-tests-reusable-workflow.yaml" isn't being recognized as a reusable workflow. Is there something on the Github side that we have to enable before using this feature? Thanks in advance!

Reply:
Thank you! It was able to recognize it! Will continue to make changes now.